### DISCUSSION TITLE
Please, explain config.courier.smtp.connection_uri more detailed

### DISCUSSION THREAD

**Discussion by victorvat** [ory/kratos#1487] [Q&A] [answered]
Hi

I'd like to get more detailed explanation of each parts in `config.courier.smtp.connection_uri`
I guess, it should be like
`<protocol>://<user>:<password>@<server>:<port>/<parameters>`
where 

- **protocol** is: _smtp_ or _smtps_ or _imap_ or _pop3_ etc

- **user** is a registered username on a email-server. (How about names like _foo\bar_ or _foo@bar_ ? )

- **password** is a passwod for a registered username on a email-server. (How about some specials in password like percent or backslash ?)

- **port** is a port where email-server is listening on (🤔not related to kratos: how to change this port number in _mailslurper_ )

- **parameters**: at the moment `?skip_ssl_verify=true` or `?skip_ssl_verify=false`

🙏

**Comment by zepatrik** [comment, marked as answer]
This depends on the service you use, e.g. for mailslurper look into [their docs](https://github.com/mailslurper/mailslurper/wiki) or for mailgun look into [their docs](https://documentation.mailgun.com/en/latest/user_manual.html#smtp-protocol). Similar to how the DSN is used to connect to the database, the SMTP URI depends on the upstream service.
